lib/tile-analyze.js

```diff
@@ -64,18 +64,32 @@ module.exports = function(filePath, options) {
 
   function analyzeSourceStream(source) {
     return new Promise((resolve, reject) => {
-      const zxyStream = source.createZXYStream();
+      let count = 1;
+
+      const zxyStream = source.createZXYStream({ maxTiles: 10 });
       const readStream = tilelive.createReadStream(source, { type: 'list' });
+      const analysisStream = new TileAnalyzeStream(analyzeTile);
+
       zxyStream.on('error', reject)
         .pipe(readStream)
-        .pipe(new TileAnalyzeStream(analyzeTile))
+        .pipe(analysisStream)
         .on('error', reject)
         .on('end', () => {
           resolve(Object.assign(stats, {
             layers: _.values(layerMap),
           }));
         })
         .resume();
+
+      // kill the stream when we've hit a maximum number of tiles
+      readStream.on('data', function() {
+        if (count >= options.maxTiles) {
+          readStream.unpipe(analysisStream);
+          readStream.end();
```

Comment:
Nice. Properly closing streams is typically a big 👀 ❓ for me , but this looks super clear and👌